### PR TITLE
Aditya 01 actuators

### PR DESCRIPTION
**Pull request by adityaguptakd**

### CODE REVIEW

File: ceaos/actuators/actuator_definition.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        raise NotImplementedError
+
+    def is_running(self):
+        return self.running
+
+    def calibrate(self):
+        """
+        This method is used to calibrate the actuator. Need not be implemented for actuators that don't require calibration
+        """
+        pass
```
Comment (nandantumu) [inline ceaos/actuators/actuator_definition.py:29]: Remember the newline at the end of the file! Use `flake8` to catch syntax/style issues

File: ceaos/actuators/actuator_definition.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+"""
+This file contains the basic interface for sensors in this code base.
+"""
+
+
+class Actuator:
+    def __init__(self) -> None:
+        raise NotImplementedError
+
+    def initialize(self):
```
Comment (nandantumu) [inline ceaos/actuators/actuator_definition.py:10]: Why split out the initialize method from `__init__`?

File: ceaos/sensors/temp_humidity.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		if self.s.humidity() != None and self.s.temperature() != None:
+			return ('{:3.2f}'.format(self.s.humidity() / 1.)), ('{:3.2f}'.format(self.s.temperature() / 1.))
+		else:
+			return "error"
+
+	def calibrate(self, calib_val_humidity, calib_val_temperature):
+		humidity, temperature = self.read_value()
+		self.calib_humidity = calib_val_humidity - humidity
+		self.calib_temperature = calib_val_temperature - temperature
+		pass
```
Comment (nandantumu) [inline ceaos/sensors/temp_humidity.py:24]: Newline!

File: ceaos/sensors/PiCamera.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		date_str = now.strftime("%m_%d_%Y__%H:%M")
+		self.camera.start_preview()
+		self.camera.annotate_text = date_str
+		sleep(3)
+		self.camera.capture('/home/pi/Documents/Plant_Photos/%s.jpg' % date_str)
+		self.camera.stop_preview()
+
+	def calibrate(self, rotation):
+		#rotate camera with this
+		self.camera.rotation = rotation
```
Comment (nandantumu) [inline ceaos/sensors/PiCamera.py:23]: This is great overall, easy to understand, clear code.

File: ceaos/sensors/PiCamera.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+		self.camera = PiCamera()
+		self.camera.rotation = 180
+
+	def read_value(self):
+		now = datetime.now()
+		date_str = now.strftime("%m_%d_%Y__%H:%M")
+		self.camera.start_preview()
+		self.camera.annotate_text = date_str
+		sleep(3)
+		self.camera.capture('/home/pi/Documents/Plant_Photos/%s.jpg' % date_str)
```
Comment (nandantumu) [inline ceaos/sensors/PiCamera.py:18]: Should we let this be configurable too? From `__init__`?

File: ceaos/sensors/PiCamera.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+	def __init__(self) -> None:
+		self.camera = PiCamera()
+		self.camera.rotation = 180
+
+	def read_value(self):
+		now = datetime.now()
+		date_str = now.strftime("%m_%d_%Y__%H:%M")
+		self.camera.start_preview()
+		self.camera.annotate_text = date_str
+		sleep(3)
```
Comment (nandantumu) [inline ceaos/sensors/PiCamera.py:17]: Is the sleeping to account for a camera quirk?

File: tests/sensors/artificial_sensor_test.py [hunk 7]
```diff
@@ -1,27 +0,0 @@
...
-def test_value_set():
-    a = setup_sensor()
-    a.set_value(25.0)
-    assert (a.value == 25.0)
-
-
-def test_value_exception():
-    a = setup_sensor()
-    with pytest.raises(ValueError):
-        a.set_value("A")
```
Comment (nandantumu) [inline tests/sensors/artificial_sensor_test.py:27]: Why delete this test?

**Review by nandantumu — CHANGES_REQUESTED** [review]
Looks like its going in the right direction, some style things, and also missing tests!

File: tests/sensors/artificial_sensor_test.py (hunk 7 above)
Comment (adityaguptakd) [reply tests/sensors/artificial_sensor_test.py:27]: wasn't even aware I deleted it, let me fix that

File: ceaos/sensors/PiCamera.py (hunk 5 above)
Comment (adityaguptakd) [reply ceaos/sensors/PiCamera.py:18]: yeah I think that would be better, let me change that

**Pull request closed without merging**